src/test/java/seedu/address/model/position/UniquePositionListTest.java

```diff
@@ -0,0 +1,173 @@
+        assertTrue(uniquePositionList.contains(editedAdminAssistant));
+    }
+
+    @Test
+    public void add_nullPosition_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> uniquePositionList.add(null));
+    }
+
+    @Test
+    public void add_duplicatePosition_throwsDuplicatePositionException() {
+        uniquePositionList.add(ADMIN_ASSISTANT);
+        assertThrows(DuplicatePositionException.class, () -> uniquePositionList.add(ADMIN_ASSISTANT));
+    }
+
+    @Test
+    public void setPosition_nullTargetPerson_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> uniquePositionList.setPosition(null, ADMIN_ASSISTANT));
+    }
+
+    @Test
+    public void setPosition_nullEditedPerson_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () ->
+                uniquePositionList.setPosition(ADMIN_ASSISTANT, null));
+    }
+
+    @Test
+    public void setPosition_targetPersonNotInList_throwsPositionNotFoundException() {
+        assertThrows(PositionNotFoundException.class, () ->
+                uniquePositionList.setPosition(ADMIN_ASSISTANT, ADMIN_ASSISTANT));
+    }
+
+    @Test
+    public void setPosition_editedPersonIsSamePerson_success() {
+        uniquePositionList.add(ADMIN_ASSISTANT);
+        uniquePositionList.setPosition(ADMIN_ASSISTANT, ADMIN_ASSISTANT);
+        UniquePositionList expectedUniquePositionList = new UniquePositionList();
+        expectedUniquePositionList.add(ADMIN_ASSISTANT);
+        assertEquals(expectedUniquePositionList, uniquePositionList);
+    }
+
+    @Test
+    public void setPosition_editedPersonHasSameIdentity_success() {
+        uniquePositionList.add(ADMIN_ASSISTANT);
+        Position editedAdminAssistant = new PositionBuilder(ADMIN_ASSISTANT).build();
+        uniquePositionList.setPosition(ADMIN_ASSISTANT, editedAdminAssistant);
+        UniquePositionList expectedUniquePositionList = new UniquePositionList();
+        expectedUniquePositionList.add(editedAdminAssistant);
+        assertEquals(expectedUniquePositionList, uniquePositionList);
+    }
+
+    @Test
+    public void setPosition_editedPersonHasDifferentIdentity_success() {
+        uniquePositionList.add(ADMIN_ASSISTANT);
+        uniquePositionList.setPosition(ADMIN_ASSISTANT, BOOKKEEPER);
+        UniquePositionList expectedUniquePositionList = new UniquePositionList();
+        expectedUniquePositionList.add(BOOKKEEPER);
+        assertEquals(expectedUniquePositionList, uniquePositionList);
+    }
+
+    @Test
+    public void setPosition_editedPersonHasNonUniqueIdentity_throwsDuplicatePositionException() {
+        uniquePositionList.add(ADMIN_ASSISTANT);
+        uniquePositionList.add(BOOKKEEPER);
+        assertThrows(DuplicatePositionException.class, () ->
+                uniquePositionList.setPosition(ADMIN_ASSISTANT, BOOKKEEPER));
+    }
+
+    @Test
+    public void remove_nullPerson_throwsNullPointerException() {
```

Comment:
perhaps "person" should be "position" here?

Reply:
Changes made accoordinly.